src/main/java/org/opensearch/knn/index/codec/KNN87Codec/KNN87Codec.java

```diff
@@ -5,124 +5,66 @@
 
 package org.opensearch.knn.index.codec.KNN87Codec;
 
+import org.apache.lucene.codecs.FilterCodec;
+import org.apache.lucene.codecs.lucene87.Lucene87Codec;
 import org.opensearch.knn.index.codec.KNN80Codec.KNN80CompoundFormat;
 import org.opensearch.knn.index.codec.KNN80Codec.KNN80DocValuesFormat;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
 import org.apache.lucene.codecs.Codec;
 import org.apache.lucene.codecs.CompoundFormat;
 import org.apache.lucene.codecs.DocValuesFormat;
-import org.apache.lucene.codecs.FieldInfosFormat;
-import org.apache.lucene.codecs.LiveDocsFormat;
-import org.apache.lucene.codecs.NormsFormat;
-import org.apache.lucene.codecs.PointsFormat;
-import org.apache.lucene.codecs.PostingsFormat;
-import org.apache.lucene.codecs.SegmentInfoFormat;
-import org.apache.lucene.codecs.StoredFieldsFormat;
-import org.apache.lucene.codecs.TermVectorsFormat;
 import org.apache.lucene.codecs.perfield.PerFieldDocValuesFormat;
 
 /**
  * Extends the Codec to support a new file format for KNN index
  * based on the mappings.
  *
  */
-public final class KNN87Codec extends Codec {
+public final class KNN87Codec extends FilterCodec {
 
-    private static final Logger logger = LogManager.getLogger(KNN87Codec.class);
     private final DocValuesFormat docValuesFormat;
     private final DocValuesFormat perFieldDocValuesFormat;
     private final CompoundFormat compoundFormat;
-    private Codec lucene87Codec;
-    private PostingsFormat postingsFormat = null;
 
     public static final String KNN_87 = "KNN87Codec";
-    public static final String LUCENE_87 = "Lucene87"; // Lucene Codec to be used
 
+    /**
+     * No arg constructor that uses Lucene87 as the delegate
+     */
     public KNN87Codec() {
-        super(KNN_87);
+        this(new Lucene87Codec());
+    }
+
+    /**
+     * Constructor that takes a Codec delegate to delegate all methods this code does not implement to.
+     *
+     * @param delegate codec that will perform all operations this codec does not override
+     */
+    public KNN87Codec(Codec delegate) {
+        super(KNN_87, delegate);
         // Note that DocValuesFormat can use old Codec's DocValuesFormat. For instance Lucene84 uses Lucene80
         // DocValuesFormat. Refer to defaultDVFormat in LuceneXXCodec.java to find out which version it uses
-        this.docValuesFormat =  new KNN80DocValuesFormat();
+        this.docValuesFormat =  new KNN80DocValuesFormat(delegate.docValuesFormat());
         this.perFieldDocValuesFormat = new PerFieldDocValuesFormat() {
             @Override
             public DocValuesFormat getDocValuesFormatForField(String field) {
+                // If the delegate is an instanceof Lucene87, we can get a field specific doc values format
+                // delegate
+                if (delegate instanceof Lucene87Codec) {
```

Comment:
Are we not doing this for every code >= Lucene87? Wondering why this should be only specific to Lucene87

Reply:
Right, so Codec does not have the method `getDocValuesFormatForField`. Lucene8XCodec's do, however. So, I felt like it made sense to map this to the LuceneCodec version our KNN codec version maps to.

Reply:
We can pull this to the separate DocValuesFormat class and inject to knn codec during construction time. We can also make getDocValuesFormatForField part of our codec facade abstraction, this should help to decrease coupling between our and lucene classes. These ideas are more for next refactoring.

Reply:
@martin-gaievski what would this look like?

Reply:
I actually was able to remove this code. The reason beingthat the delegate could have the per field functionality and we dont need to do anything to get it.

Reply:
nice. My initial idea was to create a facade interface for knn codec and then inject it to KNNCodecService using DI similar to spring or guice. I may try to play with it while working on integration with opensearch 2.0/lucene 9.1, we need a codec based on lucene 9.1